modules/Markbook/markbook_edit_addMultiProcess.php

```diff
@@ -39,8 +39,8 @@
         $name = $_POST['name'] ?? '';
         $description = $_POST['description'] ?? '';
         $type = $_POST['type'] ?? '';
-        $date = (!empty($_POST['date']))? dateConvert($guid, $_POST['date']) : date('Y-m-d');
-        $gibbonSchoolYearTermID = (!empty($_POST['gibbonSchoolYearTermID']))? $_POST['gibbonSchoolYearTermID'] : null;
+        $date = dateConvert($guid, $_POST['date']) ?? date('Y-m-d');
```

Comment:
This is not equivalent to the original line:

```php
$date = (!empty($_POST['date']))? dateConvert($guid, $_POST['date']) : date('Y-m-d');
```

In the original code, if `$_POST['date']` is not set, the `$date` will be assigned `date('Y-m-d')`. Whereas, in the replacement:

```php
$date = dateConvert($guid, $_POST['date']) ?? date('Y-m-d');
```

if `$_POST['date']` is not set, PHP will still always try to evaluate it as `null` (with "*Notice: Undefined index: date in ...*") then pass to `dateConvert()` for value. And since `dateConvert` always return something, the fallback would never work.

I don't believe you can rewrite this with null coalesce here. As far as I can see, the best option would be to change it back to the original version.